minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/service/PeriodicStatusReporterManager.java

```diff
@@ -0,0 +1,130 @@
+            try {
+                periodicStatusReporter.stop();
+            } catch (Exception exception) {
+                LOGGER.error("Could not successfully stop periodic status reporter " + periodicStatusReporter.getClass() + " due to ", exception);
+            }
+        }
+    }
+
+    public FlowStatusReport statusReport(String statusRequest) {
+        MiNiFiStatus status = miNiFiStatusProvider.getStatus(miNiFiParameters.getMiNiFiPort(), miNiFiParameters.getMinifiPid());
+
+        List<String> problemsGeneratingReport = new LinkedList<>();
+        if (!status.isProcessRunning()) {
+            problemsGeneratingReport.add("MiNiFi process is not running");
+        }
+
+        if (!status.isRespondingToPing()) {
+            problemsGeneratingReport.add("MiNiFi process is not responding to pings");
+        }
+
+        if (!problemsGeneratingReport.isEmpty()) {
+            FlowStatusReport flowStatusReport = new FlowStatusReport();
+            flowStatusReport.setErrorsGeneratingReport(problemsGeneratingReport);
+            return flowStatusReport;
+        }
+
+        return getFlowStatusReport(statusRequest, status.getPort());
+    }
+
+    private Set<PeriodicStatusReporter> initializePeriodicNotifiers() {
+        LOGGER.debug("Initiating bootstrap periodic status reporters...");
+        Set<PeriodicStatusReporter> statusReporters = new HashSet<>();
+
+        String reportersCsv = bootstrapProperties.getProperty(STATUS_REPORTER_COMPONENTS_KEY);
+
+        if (reportersCsv != null && !reportersCsv.isEmpty()) {
+            for (String reporterClassname : reportersCsv.split(",")) {
+                try {
+                    Class<?> reporterClass = Class.forName(reporterClassname);
+                    PeriodicStatusReporter reporter = (PeriodicStatusReporter) reporterClass.newInstance();
+                    reporter.initialize(bootstrapProperties, this);
+                    statusReporters.add(reporter);
+                    LOGGER.debug("Initialized {} notifier", reporterClass.getCanonicalName());
+                } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+                    throw new RuntimeException("Issue instantiating notifier " + reporterClassname, e);
+                }
+            }
+        }
+        return statusReporters;
+    }
+
+    private FlowStatusReport getFlowStatusReport(String statusRequest, int port) {
+        FlowStatusReport flowStatusReport;
+        try {
+            flowStatusReport = miNiFiCommandSender.sendCommandForObject(FLOW_STATUS_REPORT_CMD, port, FlowStatusReport.class, statusRequest);
+        } catch (Exception e) {
+            flowStatusReport = new FlowStatusReport();
+            String message = "Failed to get status report from MiNiFi due to:" + e.getMessage();
+            flowStatusReport.setErrorsGeneratingReport(Collections.singletonList(message));
+            LOGGER.error(message);
```

Comment:
Recommend logging exception for tracking:
```suggestion
            LOGGER.error(message, e);
```

Reply:
Added